File: VB/AnimateImageComboBox/MyImageComboboxEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors.Registrator
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace AnimateImageComboBox
	 <UserRepositoryItem("Register")> _
	 Friend Class MyImageComboboxEdit
		 Inherits ImageComboBoxEdit
		 Shared Sub New()
			RepositoryItemMyImageComboBox.Register()
		 End Sub

			Public Sub New()
			End Sub

		Public Overridable Sub StartAnimation()
			ViewInfo.StartAnimation()
		End Sub
		Public Overridable Sub StopAnimation()
			ViewInfo.StopAnimation()
		End Sub

		Protected Friend Shadows ReadOnly Property ViewInfo() As CustomImageComboBoxEditViewInfo
			Get
				Return TryCast(MyBase.ViewInfo, CustomImageComboBoxEditViewInfo)
			End Get
		End Property

		Public Overrides ReadOnly Property EditorTypeName() As String
				Get
					Return RepositoryItemMyImageComboBox.EditorName
				End Get
		End Property


		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyImageComboBox
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyImageComboBox)
			End Get
		End Property

		Protected Overrides Sub OnVisibleChanged(ByVal e As EventArgs)
			MyBase.OnVisibleChanged(e)
			If ((Not Visible)) AndAlso (ViewInfo.CanImagesBeAnimated) Then
				StopAnimation()
			ElseIf Visible AndAlso (ViewInfo.CanImagesBeAnimated) Then
				StartAnimation()
			End If

		End Sub

	 End Class











End Namespace

File: CS/AnimateImageComboBox/MyImageComboboxEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Registrator;
using System.ComponentModel;
using DevExpress.XtraEditors;

namespace AnimateImageComboBox {
     [UserRepositoryItem("Register")]
    class MyImageComboboxEdit: ImageComboBoxEdit{
         static MyImageComboboxEdit() {
            RepositoryItemMyImageComboBox.Register();
        }

            public MyImageComboboxEdit() {
        }

        public virtual void StartAnimation() { ViewInfo.StartAnimation(); }
        public virtual void StopAnimation() { ViewInfo.StopAnimation(); }

        protected internal new CustomImageComboBoxEditViewInfo ViewInfo { get { return base.ViewInfo as CustomImageComboBoxEditViewInfo; } }

        public override string EditorTypeName {
                get { return RepositoryItemMyImageComboBox.EditorName; }
        }


        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]

        public new RepositoryItemMyImageComboBox Properties {
            get { return base.Properties as RepositoryItemMyImageComboBox; }
        }

        protected override void OnVisibleChanged(EventArgs e) {
            base.OnVisibleChanged(e);
            if ((!Visible)&& (ViewInfo.CanImagesBeAnimated) ) StopAnimation();
            else if (Visible && (ViewInfo.CanImagesBeAnimated)) StartAnimation();
            
        }

    }


  






         
     
}
